### PR TITLE
kio-gdrive: init at version 1.2.7

### PR DESCRIPTION
**Pull request by konstantinous**
I had to add signon, signon-ui, kaccounts-integration and providers
packages to be able to build kio-gdrive ( one of the packages from #62168 )

Package works well on my laptop. Signond daemon is probably not able to find it's plugins installed into user profile. However, it doesn't affect kio-gdrive itself.

###### Motivation for this change
Add google drive integration to KDE dolphin and other kio aware apps.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/kde/kaccounts-integration.nix [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+  ];
+
+  meta = with stdenv.lib; {
+    description = "Small system to administer web accounts for the sites and services across the KDE desktop";
+    homepage = https://github.com/KDE/kaccounts-integration;
+    platforms = platforms.linux;
+
+    license = with licenses; [
+      gpl2
+      lgpl2.1
```
Comment (ttuegel) [inline pkgs/applications/kde/kaccounts-integration.nix]:
```suggestion
      lgpl21
```

File: pkgs/applications/kde/signon-kwallet-extension.nix [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+  signond
+}:
+
+mkDerivation rec {
+
+  name = "signon-kwallet-extension";
+
+  preConfigure = ''
+    # mock signond plugins dir
+    sed 's/set(SIGNONEXTENSION_PLUGINDIR ''${_pkgconfig_invoke_result})/set(SIGNONEXTENSION_PLUGINDIR lib\/signon\/extensions )/g' -i cmake/modules/FindSignOnExtension.cmake
```
Comment (ttuegel) [inline pkgs/applications/kde/signon-kwallet-extension.nix]: If you're replacing text verbatim (not using a regular expression) it's preferred to use [substituteInPlace](https://nixos.org/nixpkgs/manual/#ssec-stdenv-functions) because `sed` does not fail if there are no matches.

File: pkgs/top-level/all-packages.nix [hunk 3]
```diff
@@ -13600,6 +13600,12 @@ in
 
   shhopt = callPackage ../development/libraries/shhopt { };
 
+  signond = libsForQt5.callPackage ../development/libraries/signond { };
```
Comment (ttuegel) [inline pkgs/top-level/all-packages.nix]: If these derivations provide libraries, then they need to go into `mkLibsForQt5` instead of the top level.

File: pkgs/top-level/all-packages.nix [hunk 4]
```diff
@@ -19164,6 +19170,8 @@ in
 
   kid3 = libsForQt5.callPackage ../applications/audio/kid3 { };
 
+  kio-gdrive = kdeApplications.callPackage ../applications/misc/kio-gdrive { };
```
Comment (ttuegel) [inline pkgs/top-level/all-packages.nix]:
```suggestion
  kio-gdrive = libsForQt5.callPackage ../applications/misc/kio-gdrive { };
```

**Review by ttuegel — CHANGES_REQUESTED** [review]
I have a few requests, but generally looks good! :smiley:

File: pkgs/top-level/all-packages.nix (hunk 4 above)
Comment (konstantinous) [reply pkgs/top-level/all-packages.nix]: Hello,
This package depends on some libraries from kde (e.g. libkgapi, kcalcore, kaccounts-integration). However, this package is not a part of "kde applications" set. I don't know what is the best way to provide these dependencies, but "libsForQt5.callPackage" will not work here...

File: pkgs/top-level/all-packages.nix (hunk 3 above)
Comment (konstantinous) [reply pkgs/top-level/all-packages.nix]: Fixed.

File: pkgs/applications/kde/signon-kwallet-extension.nix (hunk 2 above)
Comment (konstantinous) [reply pkgs/applications/kde/signon-kwallet-extension.nix]: Fixed

File: pkgs/top-level/all-packages.nix (hunk 4 above)
Comment (ttuegel) [reply pkgs/top-level/all-packages.nix]: The KDE Applications libraries should be added to `mkLibsForQt5` [here](https://github.com/NixOS/nixpkgs/blob/master/pkgs/top-level/all-packages.nix#L13321), it's probably just an oversight that they weren't already.

**Review by jonringer — COMMENTED** [review]
signon-ui seems to fail to build

```
Project MESSAGE: ==== NOTE: Remember to add your API headers into `headers.files' for installation!
Project ERROR: accounts-qt5 development package not found
make[1]: *** [Makefile:44: sub-signon-ui-pro-make_first] Error 3
make[1]: Leaving directory '/build/signon-ui-3acb654/src'
make: *** [Makefile:92: sub-src-make_first-ordered] Error 2
builder for '/nix/store/cx6wbs1lrm47f4jzq09bgpj56xyqs0j7-signon-ui.drv' failed with exit code 2
error: build of '/nix/store/cx6wbs1lrm47f4jzq09bgpj56xyqs0j7-signon-ui.drv' failed
```

**Comment by konstantinous** [comment]
accounts-qt5 is inside buildInputs, so the error is seems weird to me. And it works in my environment... Would you kindly give me more information so I could try to reproduce it?

File: maintainers/maintainer-list.nix [hunk 5]
```diff
@@ -3837,6 +3837,11 @@
     githubId = 15692230;
     name = "Muhammad Herdiansyah";
   };
+  konstantsky = {
+    email = "konstantin@konstantsky.pw";
+    github = "konstantinous";
+    name = "Konstantin Khokhlov";
+  };
```
Comment (jonringer) [inline maintainers/maintainer-list.nix]: should be a separate commit:
```
maintainers: add konstantsky
```
also add githubId

**Review by jonringer — COMMENTED** [review]
each package addition should be it's own commit, and modules commits are generally labeled as:
```
nixos/module: concise message
```

File: maintainers/maintainer-list.nix (hunk 5 above)
Comment (konstantinous) [reply maintainers/maintainer-list.nix]: Done.

**Review by rnhmjoj — CHANGES_REQUESTED** [review]
I'm not a KDE user, I could set up a test environment but I wouldn't know how to use this properly so I trust you saying it's working well.
On the packaging sides of things it looks good besides the bunch of small improvements that I mentioned. By the way, good job and thank you for contributing.

File: pkgs/development/libraries/accounts-qml-module/default.nix [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+  buildInputs = [ accounts-qt qtdeclarative qttools signond qtbase ];
+  nativeBuildInputs = [ pkgconfig qmake ];
+
+  qmakeFlags = [
+    "CONFIG+=no_docs"
+  ];
+
+  preConfigure = ''
+    export QT_PLUGIN_PATH="${qtbase.bin}/${qtbase.qtPluginPrefix}"
+    substituteInPlace src/src.pro --replace '$$[QT_INSTALL_BINS]' '${qtdeclarative.dev}/bin'
```
[the remark this answers is not in the document]
Comment (konstantinous) [reply pkgs/development/libraries/accounts-qml-module/default.nix]: They are not needed anymore. You are right. Probably I forgot to pull latest changes before testing.

File: pkgs/development/libraries/accounts-qt/default.nix [hunk 7]
```diff
@@ -12,15 +12,20 @@ stdenv.mkDerivation rec {
   };
 
   buildInputs = [ glib libaccounts-glib qtbase ];
-  nativeBuildInputs = [ doxygen pkgconfig qmake ];
+  nativeBuildInputs = [ doxygen pkgconfig qmake graphviz ];
 
-  preConfigure = ''
-    qmakeFlags="$qmakeFlags LIBDIR=$out/lib CMAKE_CONFIG_PATH=$out/lib/cmake"
-  '';
+#  preConfigure = ''
```
[the remark this answers is not in the document]
Comment (konstantinous) [reply pkgs/development/libraries/accounts-qt/default.nix:19]: I removed commented lines created by me. Also I added a comment to explain why other lines were commented out. Who knows, maybe they are needed to build other packages.

File: nixos/modules/services/x11/desktop-managers/plasma5.nix [hunk 8]
```diff
@@ -158,6 +158,12 @@ in
         example = "vlc";
         description = "Phonon audio backend to install.";
       };
+
+      enableOnlineAccounts = mkOption {
```
[the remark this answers is not in the document]
Comment (konstantinous) [reply nixos/modules/services/x11/desktop-managers/plasma5.nix]: It's more convenient, thanks.

File: pkgs/development/libraries/accounts-qt/default.nix (hunk 7 above)
[the remark this answers is not in the document]
Comment (rnhmjoj) [reply pkgs/development/libraries/accounts-qt/default.nix:19]: Ok, thank you.

File: pkgs/applications/kde/signon-kwallet-extension.nix [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+  signond
+}:
+
+mkDerivation rec {
+
+  name = "signon-kwallet-extension";
+
+  preConfigure = ''
+    # mock signond plugins dir
+    substituteInPlace cmake/modules/FindSignOnExtension.cmake --replace 'set(SIGNONEXTENSION_PLUGINDIR ''${_pkgconfig_invoke_result})' 'set(SIGNONEXTENSION_PLUGINDIR lib/signon/extensions)'
```
[the remark this answers is not in the document]
Comment (rnhmjoj) [reply pkgs/applications/kde/signon-kwallet-extension.nix]: Thank you

**Review by rnhmjoj — APPROVED** [review]
Thank you. It looks good to me: I don't have any other comments to make.

**Comment by rnhmjoj** [comment]
Ah, on last thing: before merging I would squash all the "refactor" commits.

**Comment by konstantinous** [comment]
Hello @rnhmjoj , thank you for your advices. I hope I did exactly what you asked for.

**Comment by rnhmjoj** [comment]
Not exactly, sorry. I meant moving the new changes to the respective `package: init at version`.
You can do this by doing

1. a soft reset to `HEAD~`, 
2. committing each change separately,
3. interactive rebase (`git rebase -i`) in which you reorder the commits and use the `fixup` action to merge the new change with the `package: init` commit.

The idea is that the history of a PR (like fixing mistakes) shouldn't be preserved when merged into master.

**Comment by konstantinous** [comment]
Is there anything else I can do to help the merge happen?

**Comment by rnhmjoj** [comment]
Sorry, I forgot about this PR: unfortunately github doesn't send notification on new commits.
The commits look good now.

**Comment by rnhmjoj** [comment]
@jonringer can you finish your review?

File: pkgs/development/libraries/signond/default.nix [hunk 10]
```diff
@@ -0,0 +1,58 @@
...
+  preConfigure = ''
+    # don't install example plugin
+    sed -e "/example/d" -i src/plugins/plugins.pro
+    #disable tests
+    sed -i -e '/^SUBDIRS/s/tests//' signon.pro
+  '';
+
+  meta = with stdenv.lib; {
+    description = "A D-Bus service which performs user authentication on behalf of its clients.";
+    homepage = https://gitlab.com/accounts-sso/signond;
```
Comment (jonringer) [inline pkgs/development/libraries/signond/default.nix:50]: rfc 45
```suggestion
    homepage = "https://gitlab.com/accounts-sso/signond";
```

File: pkgs/development/libraries/signond/default.nix [hunk 11]
```diff
@@ -0,0 +1,58 @@
...
+    platforms = platforms.linux;
+    license = with licenses; [
+      lgpl21
+    ];
```
Comment (jonringer) [inline pkgs/development/libraries/signond/default.nix:54]:
```suggestion
    license = with licenses; [ lgpl21 ];
```

File: pkgs/development/libraries/accounts-qml-module/default.nix [hunk 12]
```diff
@@ -0,0 +1,41 @@
...
+    substituteInPlace common-project-config.pri --replace '-Werror' ' '
+    substituteInPlace accounts-qml-module.pro --replace 'tests' ' '
+  '';
+
+  installPhase = ''
+    mkdir -p $out/${qtbase.qtQmlPrefix}
+    mv src/Ubuntu  $out/${qtbase.qtQmlPrefix}
+  '';
+
+  outputs = [ "out" ];
```
Comment (jonringer) [inline pkgs/development/libraries/accounts-qml-module/default.nix:32]: Pretty sure this is default behavior
```suggestion
```

File: pkgs/development/libraries/accounts-qml-module/default.nix [hunk 13]
```diff
@@ -0,0 +1,41 @@
+{ stdenv, fetchFromGitLab, pkgconfig, qmake, accounts-qt, qttools, signond, qtdeclarative, qtbase }:
+
+stdenv.mkDerivation rec {
```
Comment (jonringer) [inline pkgs/development/libraries/accounts-qml-module/default.nix:3]: Please read https://nixos.org/nixpkgs/manual/#sec-language-qt on Qt packages in nix
```suggestion
mkDerivation rec {
```

Comment (konstantinous) [reply pkgs/development/libraries/accounts-qml-module/default.nix]: Done.

File: pkgs/development/libraries/accounts-qml-module/default.nix (hunk 12 above)
Comment (konstantinous) [reply pkgs/development/libraries/accounts-qml-module/default.nix]: Removed

File: pkgs/development/libraries/signond/default.nix (hunk 10 above)
Comment (konstantinous) [reply pkgs/development/libraries/signond/default.nix]: Done.

File: pkgs/development/libraries/signond/default.nix (hunk 11 above)
Comment (konstantinous) [reply pkgs/development/libraries/signond/default.nix]: Fixed.

**Comment by bachp** [comment]
@konstantinous Are you still working on this?  I would like to help you get this merged.

**Comment by niknetniko** [comment]
> Hoever it needs a rebase and most probably an update of the mentioned components.

I believe in the kio-gdrive was added in #122040, so perhaps this is obsolete now.
